File: WebAPI/services/CloudinaryServices.js
const cloudinary = require('cloudinary').v2;

const config = require('config');

class CloudinaryServices {
    constructor() {
        cloudinary.config({
            cloud_name: config.get("DATABASE.CLOUDINARY.NAME"),
            api_key: config.get("DATABASE.CLOUDINARY.API_KEY"),
            api_secret: config.get("DATABASE.CLOUDINARY.API_SECRET")
        });
    }

    async uploadImage(imagePath="https://upload.wikimedia.org/wikipedia/commons/a/ae/Olympic_flag.jpg") {
        const options = {
            use_filename: true,
            unique_filename: false,
            overwrite: true,
        };

        try {
            // Upload the image
            const result = await cloudinary.uploader.upload(imagePath, options);
            console.log(result)
            return result.public_id;
        } catch (error) {
            console.error(error);
            return null;
        }
    }

    async destroyImage(publicId) {
        console.log(publicId)
        try {
            await cloudinary.uploader.destroy(publicId);
            return true;
        } catch(err) {
            console.log(err)
            return false;
        }

    }

    fetchUrl(publicId) {
        const url = cloudinary.url(publicId);
        return url;
    }
}

module.exports = CloudinaryServices;

File: WebAPI/services/CloudinaryServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using CloudinaryDotNet;
using CloudinaryDotNet.Actions;
using Microsoft.Extensions.Configuration;

namespace webapi_csharp.services
{
    

    public class CloudinaryServices
    {
        private readonly IConfiguration configuration;
        private readonly Account _account;
        private readonly Cloudinary _cloudinary;

        public CloudinaryServices()
        {
            var cloudName = configuration["DATABASE:CLOUDINARY:NAME"];
            var apiKey = configuration["DATABASE:CLOUDINARY:API_KEY"];
            var apiSecret = configuration["DATABASE:CLOUDINARY:API_SECRET"];

            _account = new Account(cloudName, apiKey, apiSecret);
            _cloudinary = new Cloudinary(_account);
        }
        public string uploadImage(string imagePath = "https://upload.wikimedia.org/wikipedia/commons/a/ae/Olympic_flag.jpg")
        {
            var uploadParams = new ImageUploadParams
            {
                File = new FileDescription(imagePath),
                UseFilename = true,
                UniqueFilename = false,
                Overwrite = true,
            };

            try
            {
                var uploadResult = _cloudinary.Upload(uploadParams);
                Console.WriteLine(uploadResult);
                return uploadResult.PublicId;
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                return null;
            }
        }

        public bool destroyImage(string publicId)
        {
            Console.WriteLine(publicId);
            try
            {
                var deletionParams = new DeletionParams(publicId);
                var deletionResult = _cloudinary.Destroy(deletionParams);
                return true;
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                return false;
            }
        }

        public string fetchUrl(string publicId)
        {
            var url = _cloudinary.Api.UrlImgUp.BuildUrl(publicId);
            return url;
        }
    }

}